packages/data/src/settings/reducer.js

```diff
@@ -55,7 +64,8 @@ const receiveSettings = (
 						data: state[ group ] ? [ ...state[ group ].data, ...groupIds ] : groupIds,
 						error,
 						lastReceived: time,
-						persisting,
+						persisting: state[ group ] ? state[ group ].persisting : false,
+						dirty: state[ group ] ? state[ group ].dirty : [],
```

Comment:
This potentially sets the `dirty` property to undefined instead of to an empty array.  So instead, it should probably be something like:

```
dirty: state[ group ] && state[ group ].dirty ?
```